gourmet/tests/test_plugin_loader.py

```diff
@@ -1,37 +1,45 @@
 import unittest
 import tempfile
 import os
+import shutil
 
-import gourmet.gglobals
+from gourmet import gglobals
 
-# clear out Gourmet's DB
-tmpdir = tempfile.mktemp()
-os.makedirs(tmpdir)
-gourmet.gglobals.gourmetdir = tmpdir
 
-import gourmet.GourmetRecipeManager
-import gourmet.backends.db
+class Test (unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        # create a temporary test directory
+        cls.tmp_dir = tempfile.mkdtemp(".pl")
+        gglobals.gourmetdir = cls.tmp_dir
 
-gourmet.backends.db.RecData.__single = None
-gourmet.GourmetRecipeManager.GourmetApplication.__single = None
-# end clearing out code
+        # continue to import with 'gourmetdir' set to 'tmp_dir', otherwise 'gourmetdir' is set to '~/.gourmet'
```

Comment:
I think we should note in the comment that this attempt at isolation only really works if you're running this test module alone, not after others.

Reply:
I've updated the comment.